07_distributed_systems/01_intro/protocol/helpers.go

```diff
@@ -0,0 +1,49 @@
+package protocol
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// function appends uvarint-encoded length of item and item itself to *b slice
+// buf is a buffer for encoding uvarint
+func appendEncodedItemWithLength(b *[]byte, buf, item []byte) {
+	nBytes := binary.PutUvarint(buf, uint64(len(item)))
+	*b = append(*b, buf[:nBytes]...)
+	*b = append(*b, item...)
+}
+
+// function decodes uvarint-encoded length from current position of r
+// and returns next itemLength bytes of b
+// (r should be the reader of b[])
+// passing both b and r to function is rather dirty,
+// but it saves allocations.
+func decodeItem(b []byte, r *bytes.Reader) ([]byte, error) {
+	// abozhenko for oz:
```

Comment:
@alexbozhenko hmm I'm not sure of a more idiomatic approach... maybe @robot-dreams would know